processing/dwf_calculation_algorithm.py

```diff
@@ -0,0 +1,396 @@
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingParameterFile,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterFileDestination,
+                       QgsProcessingParameterProviderConnection,
+                       QgsProviderRegistry,
+                       QgsProviderConnectionException,
+                       QgsProcessingException)
+
+
+
+# OUTPUT : lateral as a json file 
+
+# {
+#   "offset": 180,
+#   "interpolate": false,
+#   "values": [
+#     [0.0, 0.0002],
+#     [900.0, 0.0],
+#     [1200.0, 0.0001],
+#     [1620.0, 0.0]
+#   ],
+#   "units": "m3/s",
+#   "connection_node": 4
+# }
+
+# INPUT CSV FOR API CLIENT :
+# FORMAT : lat_id, connection_node_id, timeseries
+
+# Default values
+DWF_FACTORS = [
+    [0, 0.03],
+    [1, 0.015],
+    [2, 0.01],
+    [3, 0.01],
+    [4, 0.005],
+    [5, 0.005],
+    [6, 0.025],
+    [7, 0.080],
+    [8, 0.075],
+    [9, 0.06],
+    [10, 0.055],
+    [11, 0.05],
+    [12, 0.045],
+    [13, 0.04],
+    [14, 0.04],
+    [15, 0.035],
+    [16, 0.035],
+    [17, 0.04],
+    [18, 0.055],
+    [19, 0.08],
+    [20, 0.07],
+    [21, 0.055],
+    [22, 0.045],
+    [23, 0.04],
+]
+
+
+def get_dwf_factors_from_file(file_path):
+    
+    dwf_factors = []
+    
+    with open(file_path) as csv_file:
+        reader = csv.reader(csv_file, delimiter = ',')
+        for row in reader:
+            print(row)
+            dwf_factors += [[int(row[0]), float(row[1])]]
+
+    return dwf_factors
+    
+
+
+def read_dwf_per_node(spatialite_path):
+    
+    """Obtains the 24h dry weather flow per connection node from the a 3Di model sqlite-file."""
+    conn = sqlite3.connect(spatialite_path)
+    c = conn.cursor()
+
+    # DWF per person = 120 l/inhabitant / 1000 = 0.12 m3/inhabitant
+    dwfPerPerson = 0.12
+
+    # Create empty list that holds total 24h dry weather flow per node
+    dwfPerNode24h = []
+
+    cnt = 0
+    # Create a table that contains nr_of_inhabitants per connection_node and iterate over it
+    for row in c.execute(
+        "WITH imp_surface_count AS ( SELECT impsurf.id, impsurf.nr_of_inhabitants / COUNT(impmap.impervious_surface_id) AS nr_of_inhabitants FROM v2_impervious_surface impsurf, v2_impervious_surface_map impmap WHERE impsurf.nr_of_inhabitants IS NOT NULL AND impsurf.nr_of_inhabitants != 0 AND impsurf.id = impmap.impervious_surface_id GROUP BY impsurf.id), inhibs_per_node AS ( SELECT impmap.impervious_surface_id, impsurfcount.nr_of_inhabitants, impmap.connection_node_id FROM imp_surface_count impsurfcount, v2_impervious_surface_map impmap WHERE impsurfcount.id = impmap.impervious_surface_id) SELECT ipn.connection_node_id, SUM(ipn.nr_of_inhabitants) FROM inhibs_per_node ipn GROUP BY ipn.connection_node_id"
+    ):
+        dwfPerNode24h.append([row[0], row[1] * dwfPerPerson / 3600])
+        cnt += row[1]
+
+    conn.close()
+
+    return dwfPerNode24h
```

Comment:
You're using too many empty lines to my taste :-)

Thinks of lines of code that are together as a paragraph of text. You rarely have one-line paragraphs.
You have so many one-line paragraphs that they're losing most if their meaning.